report_async/models/report_async.py

```diff
@@ -0,0 +1,162 @@
+    )
+    job_status = fields.Selection(
+        selection=[
+            ("pending", "Pending"),
+            ("enqueued", "Enqueued"),
+            ("started", "Started"),
+            ("done", "Done"),
+            ("failed", "Failed"),
+        ],
+        compute="_compute_job",
+        help="Latest Job Status",
+    )
+    job_info = fields.Text(compute="_compute_job", help="Latest Job Error Message")
+    file_ids = fields.Many2many(
+        comodel_name="ir.attachment",
+        compute="_compute_file",
+        help="List all files created by this report background process",
+    )
+
+    @api.multi
+    def _compute_job(self):
+        for rec in self:
+            rec.job_ids = (
+                self.sudo()
+                .env["queue.job"]
+                .search(
+                    [
+                        ("func_string", "like", "report.async(%s,)" % rec.id),
+                        ("user_id", "=", self._uid),
+                    ],
+                    order="id desc",
+                )
+            )
+            rec.job_status = rec.job_ids[0].sudo().state if rec.job_ids else False
+            rec.job_info = rec.job_ids[0].sudo().exc_info if rec.job_ids else False
+
+    @api.multi
+    def _compute_file(self):
+        files = self.env["ir.attachment"].search(
+            [
+                ("res_model", "=", "report.async"),
+                ("res_id", "in", self.ids),
+                ("create_uid", "=", self._uid),
+            ],
+            order="id desc",
+        )
+        for rec in self:
+            rec.file_ids = files.filtered(lambda l: l.res_id == rec.id)
+
+    def run_now(self):
+        self.ensure_one()
+        action = self.env.ref(self.action_id.xml_id)
+        result = action.read()[0]
+        ctx = safe_eval(result.get("context", {}))
+        ctx.update({"async_process": False})
+        result["context"] = ctx
+        return result
+
+    @api.multi
+    def run_async(self):
+        self.ensure_one()
+        if not self.allow_async:
+            raise UserError(_("Background process not allowed."))
+        action = self.env.ref(self.action_id.xml_id)
+        result = action.read()[0]
+        ctx = safe_eval(result.get("context", {}))
+        ctx.update({"async_process": True})
+        result["context"] = ctx
+        return result
+
+    @api.multi
+    def view_files(self):
+        self.ensure_one()
+        action = self.env.ref("report_async.action_view_files")
+        result = action.read()[0]
+        result["domain"] = [("id", "in", self.file_ids.ids)]
+        return result
+
+    @api.multi
+    def view_jobs(self):
+        self.ensure_one()
+        action = self.env.ref("queue_job.action_queue_job")
+        result = action.read()[0]
+        result["domain"] = [("id", "in", self.job_ids.ids)]
+        result["context"] = {}
+        return result
+
+    @api.model
+    @job
+    def run_report(self, docids, data, report_id, user_id):
```

Comment:
So I think... that module `base_report_to_printer` is using [Javascript](https://github.com/OCA/report-print-send/blob/10.0/base_report_to_printer/static/src/js/qweb_action_manager.js) to hijack the action of a report being printed on a certain selection of records, and changing printing behaviour based on the result of a call to `print_action_for_report_name`.

We may copy that JS code into this module and make it do a similar query for some flag that we set on the report record to specify that we want it to run asynchroneously. If that flag is True, then we notify the user with a popup that a printing job has been created and when finished, they'll get a download link in the mail. We then call some custom backend method which in turn calls this function through a cron job, like this:

```
report_async.with_delay().run_report(
     res["context"].get("active_ids", []), data, self.id, self._uid
)
```

We can then fully ignore the whole Report Center and its shenanigans and just use it in this way.